x-pack/plugins/enterprise_search/public/applications/app_search/components/synonyms/synonyms.test.tsx

```diff
@@ -5,17 +5,123 @@
  * 2.0.
  */
 
+import { setMockValues, setMockActions, rerender } from '../../../__mocks__';
+import '../../../__mocks__/shallow_useeffect.mock';
 import '../../__mocks__/engine_logic.mock';
 
 import React from 'react';
 
 import { shallow } from 'enzyme';
 
+import { EuiPageHeader, EuiButton, EuiPagination } from '@elastic/eui';
+
+import { Loading } from '../../../shared/loading';
+
+import { SynonymCard, EmptyState } from './components';
+
 import { Synonyms } from './';
 
 describe('Synonyms', () => {
+  const MOCK_SYNONYM_SET = {
+    id: 'syn-1234567890',
+    synonyms: ['a', 'b', 'c'],
+  };
+
+  const values = {
+    synonymSets: [MOCK_SYNONYM_SET, MOCK_SYNONYM_SET, MOCK_SYNONYM_SET],
+    meta: { page: { current: 1 } },
+    dataLoading: false,
+  };
+  const actions = {
+    loadSynonyms: jest.fn(),
+    onPaginate: jest.fn(),
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    setMockValues(values);
+    setMockActions(actions);
+  });
+
   it('renders', () => {
-    shallow(<Synonyms />);
-    // TODO: Check for Synonym cards, Synonym modal
+    const wrapper = shallow(<Synonyms />);
+
+    expect(wrapper.find(SynonymCard)).toHaveLength(3);
+    // TODO: Check for synonym modal
+  });
+
+  it('renders a create action button', () => {
+    const wrapper = shallow(<Synonyms />)
+      .find(EuiPageHeader)
+      .dive()
+      .children()
+      .dive();
+
+    wrapper.find(EuiButton).simulate('click');
+    // TODO: Expect open modal action
+  });
+
+  it('renders an empty state if no synonyms exist', () => {
+    setMockValues({ ...values, synonymSets: [] });
+    const wrapper = shallow(<Synonyms />);
+
+    expect(wrapper.find(EmptyState)).toHaveLength(1);
+  });
+
+  describe('loading', () => {
+    it('renders a loading state on initial page load', () => {
+      setMockValues({ ...values, synonymSets: [], dataLoading: true });
+      const wrapper = shallow(<Synonyms />);
+
+      expect(wrapper.find(Loading)).toHaveLength(1);
+    });
+
+    it('does not render a full loading state after initial page load', () => {
```

Comment:
FYI, this UX is because:

- I didn't want a full page loading screen every time the user paginates, only on initial page load
- I'll call `loadSynonyms()` on create/update/delete as well and that will have its own separate modal loading indicators, and doesn't need a full page loading screen either